### DISCUSSION TITLE
How to fetch bytes from GET response

### DISCUSSION THREAD

**Discussion by meghashyam6** [sagebind/isahc#300] [Questions] [answered]
Currently isahc::get() returns Response<Body>. Is there a way I could get the body in bytes(&[u8])? I see macros like text() in prelude but not a byte array

**Comment by sagebind** [comment, marked as answer]
The response [`Body`](https://docs.rs/isahc/1.1.0/isahc/struct.Body.html) implements [`Read`](https://doc.rust-lang.org/stable/std/io/trait.Read.html), so you can simply use the normal methods provided by the standard library to read the stream into a buffer:

```rust
let mut response = isahc::get("http://example.org")?;
let mut bytes = Vec::new();

// Read the whole response into memory.
response.body_mut().read_to_end(&mut bytes)?;
```

**Reply by meghashyam6** [reply]
That was blazing fast response. Thank you!
